pkg/provider/acme/local_store.go

```diff
@@ -111,6 +120,15 @@ func (s *LocalStore) listenSaveAction() {
 	})
 }
 
+// unSafeCopyOfStoredData creates maps copy of source. Is not thread safe, you should use `s.lock`.
+func (s *LocalStore) unSafeCopyOfStoredData(source map[string]*StoredData) map[string]*StoredData {
+	result := map[string]*StoredData{}
+	for k, v := range source {
+		result[k] = v
+	}
+	return result
+}
+
```

Comment:
Hello @walkline,

Thanks for your contribution and interest in Traefik!

I would take advantage of your method to make it thread safe.
It would be clearer i think.

```suggestion
// copyOfStoredData creates maps copy of source.
func (s *LocalStore) copyOfStoredData() map[string]*StoredData {
	s.lock.RLock()
	defer s.lock.RUnLock()
	
	result := map[string]*StoredData{}
	for k, v := range s.storedData {
		result[k] = v
	}

	return result
}

```

Reply:
Hey @rtribotte. Yeah, with your suggestion, it looks clearer, but it becomes a little bit tricky. Before calling `copyOfStoredData()`, you need to make sure that our mutex unlocked. In our case, it locked, so we will, probably, have a deadlock.